website/blog/2024-04-22-0.74-yoga-improvements-preparing-for-the-new-architecture.md

```diff
@@ -0,0 +1,441 @@
+
+```objectivec
+- (void)  userNotificationCenter:(UNUserNotificationCenter *)center
+  didReceiveNotificationResponse:(UNNotificationResponse *)response
+           withCompletionHandler:(void (^)(void))completionHandler
+{
+  // This condition passes if the notification was tapped to launch the app
+  if ([response.actionIdentifier isEqualToString:UNNotificationDefaultActionIdentifier]) {
+  // Allow the notification to be retrieved on the JS side using getInitialNotification()
+    [RCTPushNotificationManager setInitialNotification:response.notification];
+  }
+ // This will trigger 'notification' and 'localNotification' events on PushNotificationIOS
+  [RCTPushNotificationManager didReceiveNotification:response.notification];
+  completionHandler();
+}
+```
+
+Finally, delete the following methods and adapt the logic into the callbacks above which will be called instead:
+
+1. <code>[application:didReceiveLocalNotification:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622930-application?language=objc)</code> [deprecated]
+2. <code>[application:didReceiveRemoteNotification:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623117-application?language=objc)</code> [deprecated]
+3. <code>[application:didReceiveRemoteNotification:fetchCompletionHandler:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623013-application?language=objc)</code> [not deprecated, but is superseded by the <code>UNUserNotificationCenterDelegate</code> methods]
+
+Delete any usages of <code>[application:didRegisterUserNotificationSettings:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623022-application?language=objc)</code> and <code>RCTPushNotificationManager</code>’s corresponding <code>didRegisterUserNotificationSettings:</code> as well.
+
+**Example:** See the RNTester <code>[AppDelegate.mm](https://github.com/facebook/react-native/blob/main/packages/rn-tester/RNTester/AppDelegate.mm)</code>.
+
+#### JS
+
+1. Remove any references to `alertAction`.
+2. Remove the `handler` argument on any calls to `removeEventListener`.
+3. Replace any usages of `repeatInterval` by firing multiple notifications using `fireDate` or `fireIntervalSeconds` instead.
+4. Note that `soundName` will be null when it is accessed on a `Notification` returned from `getScheduledLocalNotifications()` and `getDeliveredNotifications()`.
+
+</details>
+
+### Other Breaking Changes
+
+These are some of the key breaking changes in 0.74. Please consult the [full changelog](https://github.com/facebook/react-native/blob/main/CHANGELOG.md) for the complete list of breaking changes.
+
+- Remove web performance logging from `GlobalPerformanceLogger` ([#41299](https://github.com/facebook/react-native/pull/41299)).
```

Comment:
This could seem scary to people using it, and the linked PR doesn't really have any information for people who might be using this what they should do instead. Did these APIs move somewhere else? Should people use a different approach? Perhaps comment on that PR with some details for readers.

Reply:
@rshest Is this something you can follow up? :)

Reply:
@huntie I suggest that we remove this line from the release notes altogether, as the change shouldn't affect the people's workflow, and it was more relevant to the internal experiments we were doing at the time.